### ISSUE TITLE
Search Api not working properly

### ISSUE THREAD

**Issue by stychu** [NadavShaar/react-grid-table#107]
I've just tried to use search feature and it seems it doesn't work as expected for the `object` records.
According to the docs :
>Let's say the field's value for a cell is an object:
{ ... , fullName: {firstName: 'some-first-name', lastName: 'some-last-name'} },
Its getValue function for displaying the first and last name as a full name, would be:
getValue: ({value, column}) => value.firstName + ' ' + value.lastName
The returned value will be used for searching, sorting etc...

This seem to Work half way.
I have this object: 
```js
{
	name: 'abc'
	number: 123
	property: {
		abc: 'aaaa'
	}
}
```
```js
	  // my column definition for the property.abc getter
   	  getProperty = (val) => val.abc
	   field: 'property',
      getValue: ({ value }) => getProperty(value),
```
Search will work fine for the `name` `number` values but for the `property.abc` it will work only for the first 2 characters and then it doesn't work further unless I flatten the object so all properties are on one level and the value is accessed via direct property name instead of `getValue` function

**Comment by NadavShaar** [comment]
In order to use the `abc` property it should be:
```
getValue: ({ value: myObject }) => myObject.property.abc,
```
It doesn't work?

**Comment by stychu** [comment]
@NadavShaar  I've double checked and yes this works and my above example also works. Problem show up when there are at least 2 field props with same value.

for ex.

```js
{
	name: 'abc'
	number: 123
	property: {
		first: 'aaaa',
		second: 'zzzzz'
	}
}
```
```js
const columns = [
  {
    id: 1,
    label: 'First',
    field: 'property',
    getValue: ({ value }) => value.first,
  },
  {
    id: 2,
    label: 'Second',
    field: 'property',
    getValue: ({ value }) => value.second,
  },
];
```

I expected that if I don't pass the `field` property it should return the whole row object in the `getValue` so I could map it but it forces me to define two columns with same `field: 'property'` to access `first` and `second` and this breaks the search because there are two columns with `field: 'property'`

**Comment by stychu** [comment]
https://user-images.githubusercontent.com/9383657/204028626-0e1ec068-f268-4dd8-9cff-0ef6a8a4cacc.mov

```js
columns={[
          generateColumn({
            id: 1,
            label: 'Name',
            field: 'name',
          }),
          generateColumn({
            id: 2,
            label: 'Number',
            field: 'number',
          }),
          generateColumn({
            id: 'first',
            label: 'First',
            field: 'property',
            getValue: ({ value }) => value.first,
          }),
          generateColumn({
            id: 'second',
            label: 'Second',
            field: 'property',
            getValue: ({ value }) => value.second,
          }),
        ]}
        rows={[
          {
            name: 'test',
            number: 1111,
            property: {
              first: 'bbbb',
              second: 'ggggg',
            },
          },
          {
            name: 'abc',
            number: 123,
            property: {
              first: 'kkkkk',
              second: 'zzzzzzz',
            },
          },
        ]}
        ```

**Comment by NadavShaar** [comment]
Can you please create a live example?

**Comment by stychu** [comment]
@NadavShaar https://codesandbox.io/s/react-grid-table-usage-forked-p6j20m?file=/src/App.js
Try searching by X or Y values and see search is broken where it works fine for other column search values

**Comment by NadavShaar** [comment]
Thanks, will look into it...

**Comment by NadavShaar** [comment]
Fixed in V1.1.2
